### ISSUE TITLE
Unable to get Start & Stop Position information properly

### ISSUE THREAD

**Issue by paulatrockhill** [transistorsoft/cordova-background-geolocation-lt#702]
Plugin version: 2.11.0 beta 9
Platform: iOS
OS version: iOS 10.13.4
Device manufacturer / model: simulator iPhoneX 11.4
Cordova version (cordova -v): 8.0
Cordova platform version (cordova platform ls): android 6.3.0
browser 5.0.3
ios 4.5.4
Plugin config
cordova-android-support-gradle-release 1.2.1 "cordova-android-support-gradle-release"
cordova-background-geolocation 2.11.0-beta.9 "BackgroundGeolocation"
cordova-plugin-app-version 0.1.9 "AppVersion"
cordova-plugin-background-fetch 5.3.0 "CDVBackgroundFetch"
cordova-plugin-cocoalumberjack 0.0.4 "CocoaLumberjack"
cordova-plugin-device 2.0.1 "Device"
cordova-plugin-dialogs 2.0.1 "Notification"
cordova-plugin-ios-keychain 3.0.1 "KeyChain Plugin for Cordova iOS"
cordova-plugin-native-spinner 1.1.3 "Native Spinner"
cordova-plugin-whitelist 1.3.3 "Whitelist"

## Expected Behavior
[](url)

## Actual Behavior

[index.txt](https://github.com/transistorsoft/cordova-background-geolocation-lt/files/2090250/index.txt)

we are trying to capture the gelocation upon start & stop event.  we are getting Start Record with OnOffValue:0 (not with the extras - OnOffValue:2 as we are written getCurrentPosition in Start Event) but We are getting Stop Record with (extras) OnOffValue:2 as expected.

Also if we click Start & Stop button within 1 minute interval, we are not getting any position information as said above.  Please check the configuration value and suggest us what to be changed.

Please find the attachment of our code for your reference

## Steps to Reproduce

1.
2.
3.
4.

**Comment by christocracy** [comment]
Perhaps you're interested in the `enabledchange` event

**Comment by paulatrockhill** [comment]
hi,
i am added below code with enabledchange event.
 its not sent location data to my server via Url

BackgroundGeolocation.on('enabledchange', function(event) {
  console.log("- enabledchange, plugin is enabled? ", event.enabled);
});

but i am not getting Start and Stop Separate Records.

My Output is :
{"UserID":"2592","DeviceID":"29","device":{"serial":"unknown","framework":"Cordova","manufacturer":"Apple","available":true,"platform":"iOS","cordova":"4.5.4","uuid":"65E62DEF-83F0-4E51-BFDE-71628E646361","model":"x86_64","version":"11.4","isVirtual":true},"company_token":"username","location":[{"coords":{"speed":-1,"longitude":-122.406417,"floor":null,"latitude":37.785834000000001,"accuracy":5,"altitude_accuracy":-1,"altitude":0,"heading":-1},"extras":{"onoffvalue":"0"},"is_moving":false,"event":"motionchange","odometer":0,"uuid":"AE00A9D2-8AA8-4799-922E-F92734B3BD8A","activity":{"type":"unknown","confidence":100},"battery":{"level":-1,"is_charging":false},"timestamp":"2018-06-11T15:58:31.001Z"}],"CompanyID":"2"}


{"UserID":"2592","DeviceID":"29","device":{"serial":"unknown","framework":"Cordova","manufacturer":"Apple","available":true,"platform":"iOS","cordova":"4.5.4","uuid":"65E62DEF-83F0-4E51-BFDE-71628E646361","model":"x86_64","version":"11.4","isVirtual":true},"company_token":"username","location":[{"coords":{"speed":-1,"longitude":-122.406417,"floor":null,"latitude":37.785834000000001,"accuracy":5,"altitude_accuracy":-1,"altitude":0,"heading":-1},"extras":{"onoffvalue":"0"},"is_moving":false,"event":"motionchange","odometer":0,"uuid":"48DC4E5B-F4F3-4941-AB7E-544186565FDB","activity":{"type":"unknown","confidence":100},"battery":{"level":-1,"is_charging":false},"timestamp":"2018-06-11T16:00:08.217Z"}],"CompanyID":"2"}

**Comment by christocracy** [comment]
If *I* wanted to record the location where the plugin *stopped / started*, I would do this:
```javascript

BackgroundGeolocation.on('enabledchange', function(event) {
  BackgroundGeolocation.getCurrentPosition(function(location) {
    console.log('[getCurrentPosition] enabledchange - ', event.enabled, location);
  }, function(error) {
    console.warn('[getCurrentPosition] error: ', error);
  }, {
    extras: {
      enabledchange: event.enabled
    }
  });
});

```

**Comment by paulatrockhill** [comment]
if i include this  code, 
once  start and stop button press immediately, can i get two records.
one is enabledchange : true and enabledchange: false

**Comment by christocracy** [comment]
Yes.  Why don't you try it and see?

**Comment by paulatrockhill** [comment]
HI,

i update the code but i am not getting my expected output.
i click start and stop button with in 5 to 10  seconds difference only. my device also no move its same place .

 Actual your  solution is worked for once i click start button its capture start location. after 1 or two mins later i click Stop button its capture stop location.

but i need to get location information at the time every button click action, no need to consider any time interval for start and stop.

My Outputs is:
{"UserID":"2592","DeviceID":"29","device":{"serial":"unknown","framework":"Cordova","manufacturer":"Apple","available":true,"platform":"iOS","cordova":"4.5.4","uuid":"65E62DEF-83F0-4E51-BFDE-71628E646361","model":"x86_64","version":"11.4","isVirtual":true},"company_token":"username","location":[{"coords":{"speed":-1,"longitude":-122.406417,"floor":null,"latitude":37.785834000000001,"accuracy":5,"altitude_accuracy":-1,"altitude":0,"heading":-1},"extras":{"enabledchange":true,"onoffvalue":"0"},"is_moving":false,"odometer":0,"uuid":"F57262D7-E9C1-4F04-BE9B-8C10002BAFB0","activity":{"type":"unknown","confidence":100},"battery":{"level":-1,"is_charging":false},"timestamp":"2018-06-11T16:54:25.922Z"}],"CompanyID":"2"}


{"UserID":"2592","DeviceID":"29","device":{"serial":"unknown","framework":"Cordova","manufacturer":"Apple","available":true,"platform":"iOS","cordova":"4.5.4","uuid":"65E62DEF-83F0-4E51-BFDE-71628E646361","model":"x86_64","version":"11.4","isVirtual":true},"company_token":"username","location":[{"coords":{"speed":-1,"longitude":-122.406417,"floor":null,"latitude":37.785834000000001,"accuracy":5,"altitude_accuracy":-1,"altitude":0,"heading":-1},"extras":{"enabledchange":true,"onoffvalue":"0"},"is_moving":false,"odometer":0,"uuid":"45AD835A-8AB6-4628-BB07-21322038A3F9","activity":{"type":"unknown","confidence":100},"battery":{"level":-1,"is_charging":false},"timestamp":"2018-06-11T16:55:18.291Z"}],"CompanyID":"2"}
